### PR TITLE
plat-stm32mp1: prepare PMIC regus thru SCMI

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c [hunk 1]
```diff
@@ -436,6 +438,79 @@ static void save_power_configurations(void)
...
+
+	for (n = 0; n < ARRAY_SIZE(nsec_access_regu_name); n++) {
+		if (nsec_access_regu_name[n])
+			continue;
+
+		nsec_access_regu_name[n] = calloc(1, strlen(name_ref) + 1);
+		assert(nsec_access_regu_name[n]);
+		strcpy(nsec_access_regu_name[n], name_ref);
+		break;
+	}
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c:404]: What if there is no more room? I suggest `assert(n < ARRAY_SIZE(nsec_access_regu_name));` here.

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c [hunk 2]
```diff
@@ -436,6 +438,79 @@ static void save_power_configurations(void)
...
+
+	assert(!stm32mp_nsec_can_access_pmic_regu(name_ref));
+
+	for (n = 0; n < ARRAY_SIZE(nsec_access_regu_name); n++) {
+		if (nsec_access_regu_name[n])
+			continue;
+
+		nsec_access_regu_name[n] = calloc(1, strlen(name_ref) + 1);
+		assert(nsec_access_regu_name[n]);
+		strcpy(nsec_access_regu_name[n], name_ref);
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c:402]: `strdup()`

**Review by jforissier — COMMENTED** [review]
Reviewing "plat-stm32mp1: store PMIC regulators secure assignment".

**Comment by jforissier** [comment]
For "plat-stm32mp1: add inline descriptions in pmic drivers ":
`Acked-by: Jerome Forissier <jerome@forissier.org>`

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c (hunk 2 above)
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c:402]: ok, thanks.

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c:404]: The assert below check the regu name is registered (thus alloced).

Comment (jforissier) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c:404]: That's right, ignore my comment.

**Comment by etienne-lms** [comment]
Hint for review of commit `plat-stm32mp1: factorize FDT parsing in pmic driver`:
enable github option "Hide whitespace changes".

**Comment by etienne-lms** [comment]
Squashed reviewed commit.
Applied tag on 1st and 3rd commits.
Review still open for 2nd commit `plat-stm32mp1: factorize FDT parsing in pmic driver`.

**Comment by jforissier** [comment]
For commit "plat-stm32mp1: factorize FDT parsing in pmic driver":

A bit more painful to review as often when moving things around ;-) but LGTM:
`Acked-by: Jerome Forissier <jerome@forissier.org>`

**Comment by etienne-lms** [comment]
Thanks for the time spent.
Tag applied.

**Pull request merged**